src/muc_light/mod_muc_light_room_config.erl

```diff
@@ -91,14 +95,14 @@ take_next_kv(RawConfig, [{_KeyBin, Default, Key, _Type} | RSchema]) ->
 take_next_kv([{KeyBin, _} | _], _) ->
     {error, {KeyBin, not_found}}.
 
--spec to_binary_kv(Config :: kv(), ConfigSchema :: schema()) -> binary_kv().
-to_binary_kv(Config, ConfigSchema) ->
-    ConfigWithSchema = lists:zip(lists:sort(Config), lists:keysort(3, ConfigSchema)),
-    [{KeyBin, value2b(Val, Type)} || {{Key, Val}, {KeyBin, _Default, Key, Type}} <- ConfigWithSchema].
-
-%%====================================================================
-%% Internal functions
-%%====================================================================
+take_next_binary_kv([{Key, ValBin} | RRawConfig], [{KeyBin, _Default, Key, Type} | RSchema]) ->
+    try {value, RRawConfig, RSchema, {KeyBin, value2b(ValBin, Type)}}
+    catch _:_ -> {error, {KeyBin, type_error}}
```

Comment:
I wanted to write "Why so complicated" and I after I see the old take_next_kv doing the same.
{error, _} is not covered by tests even :)

take_next logic is kinda hard to follow, in the perfect world it should be just simple maps:merge(NewConfig, maps:merge(OldConfig, Default)) 
or something.